File: bit2bitFront/bit2bitFront/src/app/modelos/Respuesta.ts
import { Pregunta } from "./Pregunta";

export class Respuesta{

   idrpta:number;
   detallerpta: string;
   fecharpta:string;
   pregunta!: Pregunta;

   constructor(){
       this.idrpta=0;
       this.detallerpta ="";
       this.fecharpta="";
   }


}

File: bit2bitBack/src/main/java/com/hipermaxi/model/Respuesta.java
package com.hipermaxi.model;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="tb_respuesta")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Respuesta {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idrpta;

    @Column(nullable = false)
    private String detallerpta;

    @Column(nullable = false)
    @Temporal(value= TemporalType.DATE)
    @DateTimeFormat(pattern = "dd/MM/yyyy")
    private Date fecharpta;

    @OneToOne()
    @JoinColumn(name="id_pregunta")
    private Pregunta pregunta;
}
